src/encoded/types/cart.py

```diff
@@ -48,51 +54,94 @@ class Collection(Item.Collection):
         pass
 
 
-def _get_carts_by_user(request, userid):
-    request.datastore = 'database'
-    return [
-        request.resource_path(v, '')
-        for k, v in request.registry[COLLECTIONS]['cart'].items()
-        if v.properties['submitted_by'] == userid
-    ]
-
-
-def _create_cart(request, user):
+def _create_cart(request, user, name=None, identifier=None, status=None):
     carts = request.registry[COLLECTIONS]['cart']
     user_props = request.embed(request.resource_path(user), '@@object')
+    cart_name = name or '{} cart'.format(user_props['title'])
+    initial_cart = {
+        'submitted_by': str(user.uuid),
+        'status': status or 'current',
+        'name': cart_name,
+        'elements': []
+    }
+    if identifier:
+        initial_cart['identifier'] = identifier
     cart = create_item(
         carts.type_info,
         request,
-        {
-            'submitted_by': str(user.uuid),
-            'status': 'current',
-            'name': '{} cart'.format(user_props['title']),
-            'elements': []
-        }
+        initial_cart,
     )
-    return request.resource_path(cart)
+    cart_path = request.resource_path(cart)
+    return cart_path
 
 
-@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
-def get_or_create_cart_by_user(context, request):
+def _get_userid(request):
     userid = [
         p.replace('userid.', '')
         for p in request.effective_principals
         if p.startswith('userid.')
     ]
     if not userid:
         raise HTTPBadRequest()
-    else:
-        userid = userid[0]
+    return userid[0]
+
+
+def _get_user(request, userid):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid)
+    return user
+
+
+@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
+def get_or_create_cart_by_user(context, request):
+    userid = _get_userid(request)
+    user = _get_user(request, userid)
+    carts = get_cart_objects_by_user(request, userid, ['disabled', 'deleted '])
```

Comment:
Similarly should pass as keyword argument:

```python
carts = get_cart_objects_by_user(
    request,
    userid,
    blocked_statuses=['disabled', 'deleted ']
)
```

Reply:
Changed